Can no longer build ESMF with LLVM 20

Because of https://github.com/esmf-org/esmf/pull/380, it is no longer possible to build ESMF with LLVM 20.

ESMF 8.9.0b05 still builds fine, 8.9.0b08 doesn't. I used LLVM 20.1.4.

Can these changes be made in a non-destructive, backwards compatible way? It should be possible to build ESMF 8.9.0 with LLVM 20 and 21.

The error is:
```
     13308    /gpfs/actions/dh/spst-llvm-atlantis/envs/ne-clang-20.1.4/install/clang/20.1.4/openmpi-5.0.6-dkjunrg/bin/mpic++  -shared -o /gpfs/actions/dh/spst-llvm-atlantis/cache/build_stage/spack-stage-esmf-8.9.0b08-gvhjjz
              z6urixajzgk32femplxnxrcgqo/spack-src/lib/libO/Linux.llvm.64.openmpi.default/libesmftrace_preload.so preload.o preload_mpi.o wrappers.o wrappers_mpi.o   -m64 -mcmodel=small -pthread -Wl,--no-as-needed -L/gpfs/a
              ctions/dh/spst-llvm-atlantis/cache/build_stage/spack-stage-esmf-8.9.0b08-gvhjjzz6urixajzgk32femplxnxrcgqo/spack-src/lib/libO/Linux.llvm.64.openmpi.default -L/gpfs/actions/dh/spst-llvm-atlantis/envs/ne-clang-20
              .1.4/install/clang/20.1.4/parallelio-2.6.2-uzbwo2j/lib -L/gpfs/actions/dh/spst-llvm-atlantis/envs/ne-clang-20.1.4/install/clang/20.1.4/netcdf-c-4.9.2-dzg3jcg/lib -L/gpfs/actions/dh/spst-llvm-atlantis/envs/ne-c
              lang-20.1.4/install/clang/20.1.4/netcdf-fortran-4.6.1-sotq3v3/lib -Wl,-rpath,/gpfs/actions/dh/spst-llvm-atlantis/cache/build_stage/spack-stage-esmf-8.9.0b08-gvhjjzz6urixajzgk32femplxnxrcgqo/spack-src/lib/libO/
              Linux.llvm.64.openmpi.default  -Wl,-rpath,/gpfs/actions/dh/spst-llvm-atlantis/envs/ne-clang-20.1.4/install/clang/20.1.4/parallelio-2.6.2-uzbwo2j/lib -Wl,-rpath,/gpfs/actions/dh/spst-llvm-atlantis/envs/ne-clang
              -20.1.4/install/clang/20.1.4/netcdf-c-4.9.2-dzg3jcg/lib -Wl,-rpath,/gpfs/actions/dh/spst-llvm-atlantis/envs/ne-clang-20.1.4/install/clang/20.1.4/netcdf-fortran-4.6.1-sotq3v3/lib -lpioc -lnetcdf -lnetcdff -lnet
              cdf -lnetcdf -ldl -lm -lrt -lFortranDecimal -lflang_rt.runtime -lstdc++ -lm -ldl
  >> 13309    /usr/bin/ld: cannot find -lflang_rt.runtime
  >> 13310    clang++: error: linker command failed with exit code 1 (use -v to see invocation)
  >> 13311    make[2]: *** [makefile:35: tracelib_preload] Error 1
```

--- a/build_config/Linux.llvm.default/build_rules.mk
+++ b/build_config/Linux.llvm.default/build_rules.mk
@@ -239,7 +239,7 @@ ESMF_F90LINKLIBS += -lrt -lstdc++ -ldl
 ############################################################
 # Link against libesmf.a using the C++ linker front-end
 #
-ESMF_CXXLINKLIBS += -lrt -lFortranDecimal -lflang_rt.runtime -lstdc++ -lm -ldl
+ESMF_CXXLINKLIBS += $(shell $(ESMF_DIR)/scripts/libs.llvm $(ESMF_F90COMPILER) $(ESMF_F90COMPILEOPTS)) -lrt -lstdc++ -lm -ldl
 
 ############################################################
 # Linker option that ensures that the specified libraries are 

--- a/scripts/libs.llvm
+++ b/scripts/libs.llvm
@@ -1,0 +1,5 @@
+#!/bin/sh
+# this script expects the LLVM flang command (or a wrapper, e.g. mpif90)
+# and returns the LLVM flang runtime libraries that need to be linked explicitly when using clang as linker
+$* -v $ESMF_DIR/scripts/hello.f90 2>&1 | grep "\-L" | awk 'BEGIN { RS=" "}; /^-l/ ' | xargs
+rm -f hello.o a.out

